File: Classes/Premade/MASPreferences/AdvancedPreferencesViewController.m

#import "AdvancedPreferencesViewController.h"

/*
@implementation AdvancedPreferencesViewController


#pragma mark -

- (id)init
{
    return [super initWithNibName:@"AdvancedPreferencesView" bundle:nil];
}

#pragma mark -
#pragma mark MASPreferencesViewController

- (NSString *)identifier
{
    return @"AdvancedPreferences";
}

- (NSImage *)toolbarItemImage
{
    return [NSImage imageNamed:@"Sped"];
}

- (NSString *)toolbarItemLabel
{
    return NSLocalizedString(@"Widget", @"Toolbar item name for the Widget preference pane");
}

- (void) awakeFromNib
{
    
    NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
    NSString*  regpref = [defaults objectForKey:@"reg-code"];
    
    
    // Read default prefrences for member name
    if (regpref == (id)[NSNull null] || regpref.length == 0 ) {
        
        [regName setStringValue:@""];
        
    } else {
        
        [regName setStringValue:regpref];
        
        [InstallButton setTitle:@"Widgets Already Installed"];
        [InstallButton setEnabled: NO];
    }
}

@end
*/

File: Classes/Premade/MASPreferences/AdvancedPreferencesViewController.swift
//
//  AdvancedPreferencesViewController.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/8/19.
//

import Foundation
import Cocoa
import MASPreferences

class AdvancedPreferencesViewController: NSViewController, MASPreferencesViewController {
    @IBOutlet var regName: NSTextField!
    @IBOutlet var InstallButton: NSButton!
    
    init() {
        super.init(nibName: "AdvancedPreferencesView", bundle: nil)
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    //MARK: - MASPreferencesViewController
    
    var viewIdentifier: String {
        return "AdvancedPreferences"
    }
    
    var toolbarItemImage: NSImage? {
        return NSImage(named: "Sped")
    }
    
    var toolbarItemLabel: String? {
        return NSLocalizedString("Widget", comment: "Toolbar item name for the Widget preference pane")
    }
    
    //MARK: - Methods
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        let regpref = UserDefaults.standard.string(forKey: "reg-code")
        
        if regpref?.isEmpty ?? true {
            regName.stringValue = ""
        } else {
            regName.stringValue = regpref ?? ""
            InstallButton.title = "Widgets Already Installed"
            InstallButton.isEnabled = false
        }
    }
}
